### PR TITLE
Implement tests for existing components

### PR DESCRIPTION
**Pull request by lcsrinaldi**

### CODE REVIEW

File: src/components/A/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import { picterTheme } from 'src/themes';
+import configureTheme from 'src/utils/configure-theme';
+
+import A from './';
+
+A.defaultProps = configureTheme(A, picterTheme);
+
+describe('[Component] A', () => {
+  it('renders without crashing', () => {
```
Comment (mattii) [inline src/components/A/index.spec.js]: Do we need this test, if we have the following tests? I think this *renders without crashing* test is only a first *better than nothing test*.

File: src/components/A/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  it('should execute handler function', () => {
+    const fn = jest.fn();
+    const wrapper = shallow(<A onClick={fn}>Link component</A>);
+
+    wrapper.simulate('click');
+
+    expect(fn).toBeCalled();
+  });
+
+  it('match snapshot', () => {
```
Comment (mattii) [inline src/components/A/index.spec.js]: `matches snapshot`

File: src/components/Button/index.spec.js [hunk 3]
```diff
@@ -1,14 +1,134 @@
...
 describe('[Component] Button', () => {
   it('renders without crashing.', () => {
     const wrapper = shallow(<Button />);
     expect(wrapper).toHaveLength(1);
   });
+
+  describe('Snapshots', () => {
+    describe('Default button', () => {
+      it('color scheme properties normal and disabled', () => {
+        expect(shallow(<Button colorScheme="default">Default color button</Button>)).toMatchSnapshot();
```
Comment (mattii) [inline src/components/Button/index.spec.js]: I'm not sure about these tests. They seem to be extremely brittle. And I don't think it really gives you much insight, when a test breaks / a snapshot changes.

But I don't know any better solution... 😉  So if somebody else sees value in it, fine for me

Comment (lcsrinaldi) [reply src/components/Button/index.spec.js]: Yes, I know what you mean Matti, I feel the same, like, of course it's going to break if I can the color to green. But this is the most recommended way of testing the styled-components styles (not behavior).

One thing that's cool, a developer created a tool for testing styled components that writes to the snapshot all of the component css properties and if it changes it shows right away what has been changed. [Check it out](https://github.com/styled-components/jest-styled-components).

Can we try it for some time? I don't know any other unit tests for it.

File: src/components/A/index.spec.js (hunk 1 above)
Comment (lcsrinaldi) [reply src/components/A/index.spec.js]: Sure!

File: src/components/A/index.spec.js (hunk 2 above)
Comment (lcsrinaldi) [reply src/components/A/index.spec.js]: Thanks

File: src/components/Button/index.spec.js (hunk 3 above)
Comment (mattii) [reply src/components/Button/index.spec.js]: Sure! Like I said, I'm fine with it!

File: src/components/Icon/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  });
+
+  it('renders element with right color value', () => {
+    const color = '#e7e7e7';
+    const wrapper = shallow(<Icon type="done" color={color} />);
+    expect(wrapper).toHaveStyleRule('color', color);
+  });
+
+  describe('[Styled Component] IconStyled', () => {
+    it('matches snapshot with size prop set to each possible value', () => {
```
Comment (mattii) [inline src/components/Icon/index.spec.js]: I think this empty test can be removed, can't it?

File: src/components/Icon/index.styled.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import styled, { css } from 'styled-components';
+
+import { variants } from 'src/utils/theme-helper';
+
+const sizeVariants = variants('size', {
```
Comment (mattii) [inline src/components/Icon/index.styled.js]: The sizes seem not to work properly:

bildschirmfoto 2017-12-04 um 11 49 52

**Pull request merged**